pipeline/detection.py

```diff
@@ -306,6 +306,10 @@ def extract_sources(self, image):
         else:
             raise ValueError(f'Unknown extraction method "{self.pars.method}"')
 
+        # perhaps this spot will end up being better, as it catches all methods and catches both psf and sourcelist 
+        # psf._upstream_bitflag = 0   # perhaps this spot will end up being better, as it catches all methods
+        # psf._upstream_bitflag |= image.bitflag
```

Comment:
Yes. I don't see much advantages to doing this inside the specific extract methods. 
In general I think the design we use is to do only the hard work inside the sub-routines and do all the housekeeping code on the `run` function (in this case it would be `extract_sources` because of the duality of the `Detector` class).